### PR TITLE
add clang format

### PR DESCRIPTION
**Pull request by robwormald**
left the comments in about travis (just a straight copy pasta from ng2 docs)

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -46,5 +48,25 @@ gulp.task('umd', ['es5', 'es6'], function (cb) {
...
+
+gulp.task('enforce-format', function() {
+  return doCheckFormat().on('warning', function(e) {
+    console.log("ERROR: You forgot to run clang-format on your change.");
+    console.log("See https://github.com/angular/angular/blob/master/DEVELOPER.md#formatting");
+    process.exit(1);
+  });
+});
+
+gulp.task('build', ['clean', 'umd', 'check-format']);
```
Comment (Foxandxss) [inline gulpfile.js]: should this check before doing shit?

Comment (robwormald) [inline gulpfile.js]: the `check-format` task will currently *warn* you but not fail the build (and give you a hint on how to do the fix. the `enforce-format` task should be set up on CI when its ready so builds will fail if they are unformatted. note a couple of the files are already warning :)

Comment (robwormald) [inline gulpfile.js]: so to answer your question, i dont think the order matters much.

**Comment by Foxandxss** [comment]
I merged it. Changed the url you point out in the `enforce-format` (It was even wrong in the angular repo, sent a PR).

Also moved the function that does the format to the bottom.

**Pull request closed without merging**